api_app/analyzers_manager/file_analyzers/filescan.py

```diff
@@ -0,0 +1,110 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class FileScan(FileAnalyzer):
+    def set_params(self, params):
+        self.upload_file = params.get("upload_file", True)
+        self.session = requests.Session()
+        self.request_url = "https://www.filescan.io/"
+        self.task_id = 0
+        self.poll_distance = 10
+        self.result = {}
+        self.max_post_tries = params.get("max_post_tries", 5)
+        self.max_get_tries = 30
+
+    def run(self):
+        binary = self.read_file_bytes()
+        if not binary:
+            raise AnalyzerRunException("File is empty")
+        self.__filescan_request_scan(binary)
+        self.__poll_status()
+        result = self.__fetch_and_build_result()
+        return result
+
+    def __filescan_request_scan(self, binary):
+        logger.info(f"Requesting Scan for: ({self.filename}), {self.md5}")
+
+        name_to_send = self.filename if self.upload_file else self.md5
+        files = {"file": (name_to_send, binary)}
+        post_sucess = False
+        for chance in range(self.max_post_tries):
```

Comment:
why is there a need of a for loop/ HTTP polling in case of POST request ? A single request should suffice.

Reply:
Sorry it was for my Debugging Purpose